_posts/2022-08-15-python-conda-environments-create.md

```diff
@@ -0,0 +1,377 @@
+
+When I was working at NEON building the NEON data skills program, the obvious
+choice for a programming language to learn and teach was R. I had used Matlab
+throughout my PhD program to process my data. But the ecology and biology 
+community which NEON was focused on were heavily using R. And it was 
+free and open source (FOSS) so I jumped ship from Matlab to R. I couldn't afford
+a Matlab license at the time anyway so it made sense. 
+
+I quickly learned to love R. It was intuitive. 
+And once I got over the pirate symbols like `<-`  and `%>%` I really  
+enjoyed using R and the `RStudio` interface is so intuitive! 
+
+## R Came With Community 
+
+But what I loved most was the community. There was SO much support and infrastructure. 
+I could to do things like - create a new package to process a specific type of data. 
+I could write vignettes for packaging. I could easily create reproducible 
+reports with citations (`RMarkdown`).
+
+Using R, I could run a workshop and send out setup instructions. People 
+installed the packages and came ready to learn. Just like that. Out of the 
+imaginary "box". Sure `GDAL` was sometimes problematic in those year working with
+R. But I figured it out. 
+
+*(Does anyone actually put software in boxes anymore?)*
+
+## Here Comes Python... and GDAL - Help!!
+
+My move to the Python programming language was much less graceful. I moved because 
+my program at CU Boulder was rooted in the Earth sciences. The earth science 
+community was using Python more regularly than R. I determined this by surveying 
+the community - public and private sector. 
+
+My surveys told me that industry was using `Python` a lot too. It's 
+generic design lends it to be used in may different fields and applications. 
+I knew that students could benefit professionally across many fields if they knew
+how to code in `Python`. 
+
+*By the way, I still think R is great for many things!
+I still use R too! No judgement here!*
+{: .notice }
+
+### How Does One Install GDAL Anyway?
+
+But, installing Python packages was much less intuitive. Quickly I had to figure 
+out whether to use `pip` or `conda` to manage installing package. And once I 
+settled on conda, it became what channel do I use? `conda-forge` seemed like 
+a good one. I found myself with environment
+conflicts. 
+
+oh yes, conflict.
+
+I'd install one thing and it would have a conflict with something else. Id remove that
+other thing and something else would break. 
+
+<figure>
+	<a href="/images/xkcd-python-environment.png">
+    <img src="/images/xkcd-python-environment.png"></a>
+	<figcaption>Environment pain when working with Python is real. However, Python
+  is an awesome programming language that is broadly used across disciplines. So 
+  it's worth figuring this all out. <i>Source xkcd.</i> 
+  </figcaption>
+</figure> 
+
+I  just wanted to code. And teach others how to code.
+
+I found myself digging around for help and trying to interpret heavily 
```

Comment:
This entire section could be tightened up... just a few sentences get to the point